ledger/participant-state/kvutils/src/main/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitter.scala

```diff
@@ -736,16 +752,12 @@ private[kvutils] class TransactionCommitter(
 }
 
 private[kvutils] object TransactionCommitter {
-
   case class DamlTransactionEntrySummary(submission: DamlTransactionEntry) {
     val ledgerEffectiveTime: Timestamp = parseTimestamp(submission.getLedgerEffectiveTime)
     val submitterInfo: DamlSubmitterInfo = submission.getSubmitterInfo
     val commandId: String = submitterInfo.getCommandId
-    val submitter: Party =
-      if (submitterInfo.getSubmittersCount == 1)
-        Party.assertFromString(submitterInfo.getSubmitters(0))
-      else
-        throw Err.InternalError("Multi-party submissions are not supported")
+    val submitters: List[Party] =
+      submitterInfo.getSubmittersList.asScala.toList.map(Party.assertFromString)
```

Comment:
You might be able to spare some collection conversions between here and L. 237

Reply:
I'm performing conversion twice because the `Engine` requires a `Set` but `authorizeSubmitters()` uses pattern matching for convenience to go recursively through all submitters. I'm not sure if changing this would really made a performance difference. What do you think?

Reply:
I don't think so, so if it's more complicated to change it than to leave it then it's fine IMO.

Reply:
I'll leave it as it is now 👍 Thank you.